src/app/process-editor/services/cardview-properties/timer-definition-item/timer-definition-item.component.ts

```diff
@@ -156,17 +156,23 @@ export class CardViewTimerDefinitionItemComponent implements OnInit, OnDestroy {
     }
 
     extractCycleFromXML(cycleDefinitionValue: string) {
-        if (cycleDefinitionValue.includes('$')) {
-            this.extractProcessVariableFromXML(cycleDefinitionValue);
-        } else if (this.isCronExpression(cycleDefinitionValue)) {
-            this.cronExpression.setValue(cycleDefinitionValue);
-            this.useCronExpression.setValue(true);
-        } else {
-            const timerDefinitions = cycleDefinitionValue.split('/');
-            this.repetitions.setValue(timerDefinitions[0].substring(1));
-
-            this.extractDateFromXML(timerDefinitions[1]);
-            this.extractDurationFromXML(timerDefinitions[2]);
+        if (cycleDefinitionValue) {
```

Comment:
Maybe it is too late to ask, but how can we have 300 loc for a new file, which we just implemented in this iteration??? Here something definitely should have been extracted already. A component should have only one responsibility, which is to handle ui interactions. I don't ask for modification, but in the future, please try to avoid long files like this.